### PR TITLE
added function to get anime id if the requested one failed

### PR DESCRIPTION
**Pull request by zezo357**
in gogo anime sometimes the anime id returned in the recent page is not right

due to gogo sometimes changing the naming schema

so i just added a function to get the right anime id from the episodes page and added example file for it 

i did not add it to the main function to avoid making the code messy/recursive i believe the example can be used in the api as a retry if the main call failed

**Did you add tests for your changes?**
No

**If relevant, did you update the documentation?**
No

**Other information**

example of wrong ids 
```
{
id: "saikyou-onmyouji-no-isekai-tenseiki",
episodeId: "saikyou-onmyouji-no-isekai-tenseiki-episode-4",
episodeNumber: 4,
title: "Saikyou Onmyouji no Isekai Tenseiki",
image: "https://gogocdn.net/cover/saikyou-onmyouji-no-isekai-tenseiki-geboku-no-youkai-domo-ni-kurabete-monster-ga-yowasugirunda-ga-1672332524.png",
url: "https://www.gogoanime.dk/saikyou-onmyouji-no-isekai-tenseiki-episode-4"
}
```
right id is https://gogoanime.tel/category/saikyou-onmyouji-no-isekai-tenseiki-geboku-no-youkai-domo-ni-kurabete-monster-ga-yowasugirunda-ga

```
{
id: "maou-gakuin-no-futekigousha-shijou-saikyou-no-maou-no-shiso-tensei-shite-shison-tachi-no-gakkou-e-kayou-ii",
episodeId: "maou-gakuin-no-futekigousha-shijou-saikyou-no-maou-no-shiso-tensei-shite-shison-tachi-no-gakkou-e-kayou-ii-episode-4",
episodeNumber: 4,
title: "Maou Gakuin no Futekigousha: Shijou Saikyou no Maou no Shiso, Tensei shite Shison-tachi no Gakkou e Kayou II",
image: "https://gogocdn.net/cover/maou-gakuin-no-futekigousha-shijou-saikyou-no-maou-no-shiso-tensei-shite-shison-tachi-no-gakkou-e-kayou-2nd-season-part-2-1672331388.png",
url: "https://www.gogoanime.dk/maou-gakuin-no-futekigousha-shijou-saikyou-no-maou-no-shiso-tensei-shite-shison-tachi-no-gakkou-e-kayou-ii-episode-4"
}
```
right id is https://gogoanime.tel/category/maou-gakuin-no-futekigousha-shijou-saikyou-no-maou-no-shiso-tensei-shite-shison-tachi-no-gakkou-e-kayou-2nd-season-part-2

### CODE REVIEW

File: src/providers/anime/gogoanime.ts [hunk 1]
```diff
@@ -260,7 +262,24 @@ class Gogoanime extends AnimeParser {
       throw new Error('Episode not found.');
     }
   };
+/**
+   *
+   * @param episodeId episode link or episode id
+   */
+fetchAnimeInfoIdFromEpisodesPage = async (episodeId: string): Promise<string> => {
```
Comment (riimuru) [inline src/providers/anime/gogoanime.ts:269]: Can you change the function name to `fetchAnimeIdFromEpisodeId`

File: src/providers/anime/gogoanime.ts [hunk 2]
```diff
@@ -242,6 +242,8 @@ class Gogoanime extends AnimeParser {
       const res = await this.client.get(episodeId);
 
       const $ = load(res.data);
+      console.log("#wrapper_bg > section > section.content_left > div:nth-child(1) > div.anime_video_body > div.anime_video_body_cate > div.anime-info > a");
+      console.log(($("#wrapper_bg > section > section.content_left > div:nth-child(1) > div.anime_video_body > div.anime_video_body_cate > div.anime-info > a").attr('href') as string).split('/')[2]);
```
Comment (riimuru) [inline src/providers/anime/gogoanime.ts:246]: Can you remove the console logs

File: examples/saikyou-onmyouji-no-isekai-tenseiki.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    return results;
+  }catch{
+    // get new id and try again (default will be episode 1 i think)
+    const anime_id=await gogoanime.fetchAnimeInfoIdFromEpisodesPage(animeId+'-episode-4');
+    const results = await gogoanime.fetchAnimeInfo(anime_id);
+    console.log(results);
+  }
+  
+
+};
```
Comment (riimuru) [inline examples/saikyou-onmyouji-no-isekai-tenseiki.ts:20]: It would be great if you could format this file

**Review by riimuru — CHANGES_REQUESTED** [review]

**Comment by zezo357** [comment]
done

**Review by riimuru — APPROVED** [review]
LGTM 👍

**Pull request merged**